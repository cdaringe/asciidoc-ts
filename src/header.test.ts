import { describe, expect, test } from "vitest";
import { toAST } from "./mod.js";

test("Parse Header (without \\n is PlainText)", async ({ expect }) => {
  const input = `= Main Header`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("Parse Header (with \\n)", async ({ expect }) => {
  const input = `== Main Header\n `;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});
