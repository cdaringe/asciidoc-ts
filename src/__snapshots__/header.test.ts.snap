// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Parse Header (with \\n) 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "Main Header",
          "type": "PlainText",
        },
      ],
      "level": 2,
      "type": "Header",
    },
  ],
  "type": "Document",
}
`;

exports[`Parse Header (without \\n is PlainText) 1`] = `
{
  "blocks": [
    {
      "content": "= Main Header",
      "type": "PlainText",
    },
  ],
  "type": "Document",
}
`;
